packages/python/plotly/plotly/express/_core.py

```diff
@@ -754,6 +754,42 @@ def apply_default_cascade(args):
         args["marginal_x"] = None
 
 
+def build_or_augment_dataframe(args, attrables, array_attrables):
+    """
+    Constructs an implicit dataframe and modifies `args` in-place.
+    `attrables` is a list of keys into `args`, all of whose corresponding
+    values are converted into columns of a dataframe.
+    Used to be support calls to plotting function that elide a dataframe
+    argument; for example `scatter(x=[1,2], y=[3,4])`.
+    """
+    if args.get("data_frame") is None:
+        df = pd.DataFrame()
+    else:
+        df = args["data_frame"]
+        df = df.reset_index()
```

Comment:
I don't think we should automatically do this, as it will flow through to the `dimension`-supporting traces :)

Reply:
yes, this has been corrected